### PR TITLE
Fix #389: SCA passes even though iProov failed

### LINKED ISSUE #389: SCA passes even though iProov failed
```
2022-09-22 17:54:12.412  INFO 2974077 --- [   scheduling-1] c.w.a.o.i.service.PresenceCheckService   : Not accepted presence check, OwnerId(activationId=7deb987f-dc81-4e20-ab1c-152fd1b2939c, userId=mockuser_2222)
2022-09-22 17:54:12.412  INFO 2974077 --- [   scheduling-1] c.w.a.o.i.service.PresenceCheckService   : Switched to PRESENCE_CHECK/REJECTED; process ID: 2fc4a823-5e95-4618-9564-0f784862cb2e
2022-09-22 17:54:12.412  INFO 2974077 --- [   scheduling-1] c.w.a.o.i.service.PresenceCheckService   : Presence check rejected, OwnerId(activationId=7deb987f-dc81-4e20-ab1c-152fd1b2939c, userId=mockuser_2222), rejectReason:'null'
2022-09-22 17:54:12.412  INFO 2974077 --- [   scheduling-1] w.a.o.i.s.IdentityVerificationOtpService : Switched to OTP_VERIFICATION/OTP_VERIFICATION_PENDING; process ID: 2fc4a823-5e95-4618-9564-0f784862cb2e
2022-09-22 17:54:12.415  INFO 2974077 --- [   scheduling-1] c.w.a.o.mock.MockOnboardingProvider      : Send OTP code called, user ID: mockuser_2222, OTP code: 72459832, resend: false
2022-09-22 17:54:13.237  INFO 2974077 --- [io-8080-exec-16] i.g.s.p.a.s.s.v3.PowerAuthServiceImpl    : GetEciesDecryptorRequest received, application key: gfv6saZiwmtOAMk9iD1xyg==, activation ID: 7deb987f-dc81-4e20-ab1c-152>
2022-09-22 17:54:13.242  INFO 2974077 --- [io-8080-exec-16] i.g.s.p.a.s.s.v3.PowerAuthServiceImpl    : GetEciesDecryptorRequest succeeded
2022-09-22 17:54:13.247  INFO 2974077 --- [io-8080-exec-21] i.g.s.p.a.s.s.v3.PowerAuthServiceImpl    : ValidateTokenRequest received, token ID: d4e5782b-013d-4830-b2ae-c8e52c46d312
2022-09-22 17:54:13.248  INFO 2974077 --- [io-8080-exec-21] i.g.s.p.a.s.s.v3.PowerAuthServiceImpl    : ValidateTokenRequest succeeded
2022-09-22 17:54:13.254  INFO 2974077 --- [io-8080-exec-14] i.g.s.p.a.s.s.v3.PowerAuthServiceImpl    : ListActivationFlagsRequest received, activation ID: 7deb987f-dc81-4e20-ab1c-152fd1b2939c
2022-09-22 17:54:13.255  INFO 2974077 --- [io-8080-exec-14] i.g.s.p.a.s.s.v3.PowerAuthServiceImpl    : ListActivationFlagsRequest succeeded
2022-09-22 17:54:33.623  INFO 2974077 --- [io-8080-exec-22] i.g.s.p.a.s.s.v3.PowerAuthServiceImpl    : GetEciesDecryptorRequest received, application key: gfv6saZiwmtOAMk9iD1xyg==, activation ID: null
2022-09-22 17:54:33.626  INFO 2974077 --- [io-8080-exec-22] i.g.s.p.a.s.s.v3.PowerAuthServiceImpl    : GetEciesDecryptorRequest succeeded
2022-09-22 17:54:37.643  INFO 2974077 --- [io-8080-exec-15] i.g.s.p.a.s.s.v3.PowerAuthServiceImpl    : GetEciesDecryptorRequest received, application key: gfv6saZiwmtOAMk9iD1xyg==, activation ID: 7deb987f-dc81-4e20-ab1c-152>
2022-09-22 17:54:37.648  INFO 2974077 --- [io-8080-exec-15] i.g.s.p.a.s.s.v3.PowerAuthServiceImpl    : GetEciesDecryptorRequest succeeded
2022-09-22 17:54:37.671  INFO 2974077 --- [io-8080-exec-13] c.w.a.o.i.s.IdentityVerificationService  : Switched to COMPLETED/ACCEPTED; process ID: 2fc4a823-5e95-4618-9564-0f784862cb2e
```

Double-check changes in #315

### PR DESCRIPTION
**Pull request by banterCZ**

### CODE REVIEW

File: enrollment-server-onboarding/src/main/java/com/wultra/app/onboardingserver/controller/api/IdentityVerificationController.java [hunk 1]
```diff
@@ -408,6 +409,37 @@ public ResponseEntity<Response> initPresenceCheck(@EncryptedRequestBody ObjectRe
...
+     * @param eciesContext ECIES context.
+     * @param apiAuthentication PowerAuth authentication.
+     * @return Presence check initialization response.
+     * @throws PowerAuthAuthenticationException Thrown when request authentication fails.
+     * @throws PowerAuthEncryptionException Thrown when request decryption fails.
+     * @throws IdentityVerificationException Thrown when identity verification is invalid.
+     */
+    @PostMapping("presence-check/submit")
+    @PowerAuthEncryption(scope = EciesScope.ACTIVATION_SCOPE)
+    @PowerAuth(resourceId = "/api/identity/presence-check/submit", signatureType = PowerAuthSignatureTypes.POSSESSION)
```
Comment (banterCZ) [inline enrollment-server-onboarding/src/main/java/com/wultra/app/onboardingserver/controller/api/IdentityVerificationController.java:425]: @kober32 A new endpoint.

File: enrollment-server-onboarding/src/main/java/com/wultra/app/onboardingserver/impl/service/IdentityVerificationOtpService.java [hunk 2]
```diff
@@ -182,13 +180,12 @@ private OtpVerifyResponse verifyPresenceCheck(final OnboardingProcessEntity proc
         final String rejectReason = idVerification.getRejectReason();
         final RejectOrigin rejectOrigin = idVerification.getRejectOrigin();
 
-        if (errorOrigin == ErrorOrigin.PRESENCE_CHECK && StringUtils.isNotBlank(errorDetail)
-                || rejectOrigin == RejectOrigin.PRESENCE_CHECK && StringUtils.isNotBlank(rejectReason)) {
+        if (errorOrigin == ErrorOrigin.PRESENCE_CHECK || rejectOrigin == RejectOrigin.PRESENCE_CHECK) {
```
Comment (banterCZ) [inline enrollment-server-onboarding/src/main/java/com/wultra/app/onboardingserver/impl/service/IdentityVerificationOtpService.java:183]: Not sure whether this is a permanent or temporary change. When we called iProov validate before verification, the reject reason was `null` so SCA passed.

File: enrollment-server-onboarding/src/main/java/com/wultra/app/onboardingserver/statemachine/StateMachineConfig.java [hunk 3]
```diff
@@ -329,14 +334,21 @@ private void configurePresenceCheckTransitions(StateMachineTransitionConfigurer<
...
                 .withExternal()
                 .source(OnboardingState.PRESENCE_CHECK_IN_PROGRESS)
-                .event(OnboardingEvent.EVENT_NEXT_STATE)
+                .event(OnboardingEvent.PRESENCE_CHECK_SUBMITTED)
+                .action(moveToPresenceCheckVerificationPendingAction)
+                .target(OnboardingState.PRESENCE_CHECK_VERIFICATION_PENDING)
+
+                .and()
+                .withExternal()
+                .source(OnboardingState.PRESENCE_CHECK_VERIFICATION_PENDING)
```
Comment (banterCZ) [inline enrollment-server-onboarding/src/main/java/com/wultra/app/onboardingserver/statemachine/StateMachineConfig.java:343]: @kober32 You may expect `PRESENCE_CHECK / VERIFICATION_PENDING` at the status endpoint.

File: enrollment-server-onboarding/src/main/java/com/wultra/app/onboardingserver/statemachine/action/presencecheck/MoveToPresenceCheckVerificationPendingAction.java [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+
+    @Override
+    @Transactional
+    public void execute(StateContext <OnboardingState, OnboardingEvent> context) {
+        final OwnerId ownerId = (OwnerId) context.getMessageHeader(EventHeaderName.OWNER_ID);
+        final IdentityVerificationEntity identityVerification = context.getExtendedState().get(ExtendedStateVariable.IDENTITY_VERIFICATION, IdentityVerificationEntity.class);
+        moveToDocumentUploadVerificationPending(ownerId, identityVerification);
+    }
+
+    private void moveToDocumentUploadVerificationPending(final OwnerId ownerId, final IdentityVerificationEntity idVerification) {
```
Comment (saalistaja) [inline enrollment-server-onboarding/src/main/java/com/wultra/app/onboardingserver/statemachine/action/presencecheck/MoveToPresenceCheckVerificationPendingAction.java:59]: I would recommend to have this in a separate service.

**Review by saalistaja — COMMENTED** [review]
Looks good, but I have not tested this.

File: enrollment-server-onboarding/src/main/java/com/wultra/app/onboardingserver/controller/api/IdentityVerificationController.java [hunk 5]
```diff
@@ -408,6 +409,37 @@ public ResponseEntity<Response> initPresenceCheck(@EncryptedRequestBody ObjectRe
...
+    public ResponseEntity<Response> submitPresenceCheck(@EncryptedRequestBody ObjectRequest<PresenceCheckInitRequest> request,
```
Comment (banterCZ) [inline enrollment-server-onboarding/src/main/java/com/wultra/app/onboardingserver/controller/api/IdentityVerificationController.java:426]: [] `PresenceCheckSubmitRequest` should be introduced

File: enrollment-server-onboarding/src/main/java/com/wultra/app/onboardingserver/statemachine/action/presencecheck/MoveToPresenceCheckVerificationPendingAction.java (hunk 4 above)
Comment (banterCZ) [reply enrollment-server-onboarding/src/main/java/com/wultra/app/onboardingserver/statemachine/action/presencecheck/MoveToPresenceCheckVerificationPendingAction.java:65]: 👍 My first reaction was: why. But I realized that similar patter is in other action classes to, so put it to one place may simplify refactor later. My dream is to change phases elsewhere.

**Review by romanstrobl — APPROVED** [review]
Nice cleanup!

**Pull request merged**